### PR TITLE
Fix attrd race condition when shutting down

### PR DESCRIPTION
**Pull request by kgaillot**

### CODE REVIEW

**Comment by kgaillot** [comment]
@clumens, can you review this when you get a chance? Thanks

File: daemons/attrd/pacemaker-attrd.h [hunk 1]
```diff
@@ -58,7 +58,7 @@ void attrd_set_requesting_shutdown(void);
 void attrd_clear_requesting_shutdown(void);
 void attrd_free_waitlist(void);
 bool attrd_requesting_shutdown(void);
```
Comment (clumens) [inline daemons/attrd/pacemaker-attrd.h:60]: Remove the reference here.

Comment (kgaillot) [reply daemons/attrd/pacemaker-attrd.h:60]: thanks

**Comment by kgaillot** [comment]
Updated for review and threw one more commit on top

File: tools/attrd_updater.c [hunk 2]
```diff
@@ -145,8 +145,11 @@ static GOptionEntry required_entries[] = {
 
 static GOptionEntry command_entries[] = {
     { "update", 'U', 0, G_OPTION_ARG_CALLBACK, command_cb,
-      "Update attribute's value in pacemaker-attrd. If this causes the value\n"
-      INDENT "to change, it will also be updated in the cluster configuration.",
+      "Update attribute's value. Required: -n/--name or -P/--pattern.\n"
+      INDENT "Optional: -d/--delay (if specified, the delay will be used if\n"
+      INDENT "the attribute needs to be created, but ignored if the\n"
+      INDENT "attribute already exists), -s/--set, -p/--private, -W/--wait,\n"
+      INDENT "-z/--utilization.\n",
```
Comment (clumens) [inline tools/attrd_updater.c:152]: You don't need to put a newline on the last line of the help output.

**Comment by clumens** [comment]
Patches look good.  ctslab is looking good through 65 runs, including multiple NearQuorumPoint runs.  I have it set to run 150 tests, so I guess we'll know tomorrow.

File: tools/attrd_updater.c (hunk 2 above)
Comment (kgaillot) [reply tools/attrd_updater.c]: thanks, updated

**Pull request merged**